herd/eventsMonad.ml

```diff
@@ -42,7 +42,9 @@ and type evt_struct = E.event_structure) =
     module VC =
       Valconstraint.Make
         (struct
-          let hexa = C.hexa let debug = C.debug.Debug_herd.solver
+          let hexa = C.hexa
+          let debug = C.debug.Debug_herd.solver
+          let keep_failed_as_undetermined = C.variant Variant.ASL_AArch64
```

Comment:
Why do we have  `-variant asl` and `-variant asl_aarch64`? Aren't both here for commanding the ASL implementation of the `AArch64` architecture?

Reply:
The ASL_AArch64 variant instruct the ASL backend to load the AArch64 shared pseudocode as an ASL library for the litmus test. This is very useful!

I reckon that it's not very explicit. Where is the canonical place to describe variants?

Reply:
The best place for documenting variants  is the interface file ``lib/variant.mli`.

I still do not understand. Assuming that  `-variant asl` and `-variant asl_aarch64` to used for executing AArch64 litmus test, why would loading the AArch64 shared pseudo-code be optional?

Reply:
Then the variant was already documented.

Both variant are meant to be used in different architectures:
 - ASL+AArch64 on the ASL arch
 - ASL on the AArch64 arch

We can maybe unify them, and find a name that is explicit in both architectures.

Reply:
Using herd with ASL architecture without the AArch64 shared pseudocode is useful when studying a very small piece of ASL code to identify its semantics.

Reply:
> Then the variant was already documented.

:)